asf_core_data/getters/supplementary_data/deprivation/imd_data.py

```diff
@@ -55,26 +32,59 @@ def get_gb_imd_data(
         "Country",
     ],
 ):
-    """Get deprivation data for England, Wales and Scotland.
+    """Get deprivation data for specific country.
 
     Args:
-        data_path (str/Path, optional): Path to ASF core data directory or 'S3'. Defaults to base_config.ROOT_DATA_PATH.
-        rel_data_path (str/Path, optional): Relative path to IMD data. Defaults to base_config.IMD_PATH.
-        usecols (list, optional):  List of features/columns to load.
-            Defaults to ["IMD Rank", "IMD Decile", "Postcode", "Income Score", "Employment Score", "Country"].
-            This selection works for all countries. None will load all features.
+        country (str, optional): Country for which to load IMD: 'England', 'Scotland', 'Wales' or 'GB'. Defaults to 'GB'.
+        data_path (str, optional): Path to IMD data in local dir or 'S3'. Defaults to "S3".
+        rel_path (_type_, optional): Relative path to IMD data. Defaults to None, leading to loading from default location.
+        usecols (list, optional): List of features/columns to load. Defaults to ["IMD Rank", "IMD Decile", "Postcode", "Income Score", "Employment Score", "Country"].
+
+    Raises:
+        IOError: Throws error if country input is not valid.
 
     Returns:
-        pandas.DataFrame:  Deprivation data for all countries.
+        pandas.DataFrame: Deprivation data.
     """
 
-    imd_path = Path(data_path) / rel_data_path
+    country = country.lower()
+    country_name = country[0].upper() + country[1:] if country != "gb" else "GB"
+
+    if country in ["gb", "all", None]:
```

Comment:
according to the docstring, country should only take one of the values: "England", "Scotland", "Wales", "GB". So if it receives "all" or None, it should yield an error. Otherwise, None and "all" should be accounted for in the docstring.

Reply:
you already have the else raising the issue below so you just need to apply the following change:
```suggestion
    if country=="gb":
```

Reply:
Yes, looks like I tried to be too accomodating. Better to keep it simple